src/lib.rs

```diff
@@ -77,6 +145,52 @@ impl VehiclePropertyValue {
     }
 }
 
+impl GetPropertyValue for EmulatorMessage {
+    fn get_value(&self) -> Result<VehiclePropertyValue> {
+        if self.value.len() != 1 {
+            return Err(VhalError::ReceiveMessageValueError);
+        }
```

Comment:
Just a suggestion/question: is it worth to be more specific, maybe even create extra error enums?
1. For no values at all.
2. For too many values. That is you could have taken the first value, but then you wouldn't know what to do with the rest, so returned an error...

Reply:
It is a tradeoff certainly, on how much fine-grained we want the errors. In this case, I think it is worth to add one specific for number of elements, yes.